common/src/main/java/com/wynntils/core/functions/expressions/ConstantExpression.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.functions.expressions;
+
+import com.wynntils.utils.type.ErrorOr;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class ConstantExpression extends Expression {
+    private static final Map<Class<?>, Function<String, Optional<Object>>> CONSTANT_EXPRESSION_PARSERS = Map.of(
+            String.class, ConstantExpression::markedStringParser,
+            Integer.class, ConstantExpression::intParser,
+            Double.class, ConstantExpression::doubleParser,
+            Boolean.class, ConstantExpression::booleanParser);
+
+    private final Object value;
+
+    protected ConstantExpression(String rawExpression, Object value) {
+        super(rawExpression);
+        this.value = value;
+    }
+
+    @Override
+    public ErrorOr<String> calculate() {
+        return ErrorOr.of(value.toString());
+    }
+
+    public static ErrorOr<Optional<Expression>> tryParse(String rawExpression) {
+        for (Function<String, Optional<Object>> value : CONSTANT_EXPRESSION_PARSERS.values()) {
+            Optional<Object> parsedValue = value.apply(rawExpression);
+
+            if (parsedValue.isPresent()) {
+                return ErrorOr.of(Optional.of(new ConstantExpression(rawExpression, parsedValue.get())));
+            }
+        }
+
+        return ErrorOr.of(Optional.empty());
+    }
+
+    // region Parsers
+
+    private static Optional<Object> markedStringParser(String rawString) {
+        if (rawString.length() > 1 && rawString.startsWith("\"") && rawString.endsWith("\"")) {
+            return Optional.of(rawString.substring(1, rawString.length() - 1));
+        }
+
+        return Optional.empty();
+    }
+
+    private static Optional<Object> intParser(String rawString) {
+        try {
+            return Optional.of(Integer.parseInt(rawString));
+        } catch (NumberFormatException e) {
+            return Optional.empty();
+        }
+    }
+
+    private static Optional<Object> doubleParser(String rawString) {
+        try {
+            return Optional.of(Double.parseDouble(rawString));
+        } catch (NumberFormatException e) {
+            return Optional.empty();
+        }
+    }
+
+    private static Optional<Object> booleanParser(String rawString) {
+        if (rawString.equalsIgnoreCase("true") || rawString.equalsIgnoreCase("false")) {
```

Comment:
maybe also allow more variants? like `t` or `f` or `1` and `0`?

Reply:
I fear that adding these will somehow make the parser less reliable. I noted this to myself, and I will get back to this. But until the full system is emerged, I will hold off on doing this.